### PR TITLE
Update advection executables to use subcell

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Since subcell currently does not support `ObserveErrorNorms`, corresponding checks were deleted from input files.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvection.hpp [hunk 1]
```diff
@@ -106,6 +125,12 @@ struct EvolutionMetavars {
   using temporal_id = Tags::TimeStepId;
   static constexpr bool local_time_stepping = false;
 
+  // The use_dg_subcell flag controls whether to use "standard" limiting (false)
+  // or a DG-FD hybrid scheme (true).
+  static constexpr bool use_dg_subcell = true;
+  static constexpr dg::Formulation dg_formulation =
```
Comment (nilsdeppe) [inline src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvection.hpp]: Why is this necessary? We control the formulation at runtime now as far as I know

File: tests/InputFiles/ScalarAdvection/Kuzmin2D.yaml [hunk 2]
```diff
@@ -25,8 +17,8 @@ DomainCreator:
   Rectangle:
     LowerBound: [0.0, 0.0]
     UpperBound: [1.0, 1.0]
-    InitialRefinement: [3, 3]
-    InitialGridPoints: [3, 3]
+    InitialRefinement: [1, 1]
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarAdvection/Kuzmin2D.yaml]: Only commenting on this input file, but switching to subcell I'd expect you increase the number of points and decrease the lev. Eg lev 3->2, points 2->4. What's the reason for decreasing both? Is it that subcell is causing the tests to time out?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
overall LGTM, please squash any changes

**Comment by yoonso0-0** [comment]
> Only commenting on this input file, but switching to subcell I'd expect you increase the number of points and decrease the lev. Eg lev 3->2, points 2->4. What's the reason for decreasing both? Is it that subcell is causing the tests to time out? 
e

File: tests/InputFiles/ScalarAdvection/Kuzmin2D.yaml (hunk 2 above)
Comment (yoonso0-0) [reply tests/InputFiles/ScalarAdvection/Kuzmin2D.yaml]: At the first place (before adding Krivodonova and Kuzmin solution) it was lev 1 with points 2, I guess this was chosen to minimize the time taken by the `execute` test running on CI. When I changed `execute` test to `execute_check_output` test in #3405 for those two types of analytic solutions I needed to increase the resolution enough to ensure that the test is catching the real error norm (starts to decrease with increasing resolution), capturing the irregular shape of those two types of initial data. Since subcell executable currently cannot perform `execute_check_output` test, I have put resolutions back to the minimal one.

**Comment by yoonso0-0** [comment]
Rebased and squashed

**Review by nilsdeppe — APPROVED** [review]
